api_app/connectors_manager/serializers.py

```diff
@@ -25,6 +68,11 @@ class ConnectorConfigSerializer(serializers.Serializer):
     secrets = serializers.JSONField()
     verification = serializers.SerializerMethodField()
 
+    def validate_secrets(self, secrets):
+        serializer = SecretSerializer(data=list(secrets.items()), many=True)
+        if serializer.is_valid(raise_exception=True):
+            return secrets
+
     def check_secrets(self, secrets):
         exceptions = {}
         for key_name, secret_dict in secrets.items():
```

Comment:
on line 79, the code is:

```python3
if "required" in secret_dict and secret_dict["required"]:
```

a better way to achieve this would be:

```python3
if secret_dict,get("required", False):
```